pkg/executables/kubectl.go

```diff
@@ -825,3 +825,116 @@ func (k *Kubectl) GetBundles(ctx context.Context, kubeconfigFile, name, namespac
 
 	return response, nil
 }
+
+func (k *Kubectl) PatchTolerations(ctx context.Context, taints []corev1.Taint, resource string, name string, kubeconfigFile string, namespace string, path string) error {
+	if len(taints) > 0 {
+		var tolerations []string
+		format := "{\"op\": \"add\", \"path\": \"%s\", \"value\":{\"key\":\"%s\",\"operator\":\"Equal\",\"value\":\"%s\",\"effect\":\"%s\"}}"
+		for _, taint := range taints {
+			if taint.Effect != "NoExecute" {
+				tolerations = append(tolerations, fmt.Sprintf(format, path, taint.Key, taint.Value, taint.Effect))
+			}
+		}
+		if len(tolerations) > 0 {
+			params := []string{
+				"patch", resource, name,
+				"--type=json", fmt.Sprintf("-p=[%s]", strings.Join(tolerations, ", ")), "-n", namespace, "--kubeconfig", kubeconfigFile,
+			}
+			_, err := k.executable.Execute(ctx, params...)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+type Toleration struct {
+	Effect   string
+	Key      string
+	Operator string
+	Value    string
+}
+
+func (k *Kubectl) UpdateTolerations(ctx context.Context, oldTaints []corev1.Taint, newTaints []corev1.Taint, resource string, name string, kubeconfigFile string, namespace string, path string) error {
+	params := []string{
+		"get", resource, name,
+		"-o", "jsonpath={range .spec.template.spec.tolerations[*]}[{@}] {end}",
```

Comment:
Why not just get json and unmarshal into a struct?
I think we would probably avoid all the string manipulation, wouldn't we?